### PR TITLE
added postgres

### PR DESCRIPTION
**Pull request by trbouma**
added postgres support

### CODE REVIEW

File: cashu/core/settings.py [hunk 1]
```diff
@@ -43,6 +43,7 @@
 MINT_URL = env.str("MINT_URL", default=None)
 MINT_HOST = env.str("MINT_HOST", default="8333.space")
 MINT_PORT = env.int("MINT_PORT", default=3338)
+MINT_DATABASE_URL = env.str("MINT_DATABASE_URL", default="data/mint")
```
Comment (callebtc) [inline cashu/core/settings.py:46]:
```suggestion
MINT_DATABASE_URL = env.str("MINT_DATABASE_URL", default=None)
```

Comment (callebtc) [inline cashu/core/settings.py:46]:
```suggestion
MINT_DATABASE = env.str("MINT_DATABASE_URL", default="data/mint")
```

File: cashu/mint/startup.py [hunk 2]
```diff
@@ -7,13 +7,13 @@
 
 from cashu.core.db import Database
 from cashu.core.migrations import migrate_databases
-from cashu.core.settings import CASHU_DIR, LIGHTNING, MINT_PRIVATE_KEY
+from cashu.core.settings import CASHU_DIR, LIGHTNING, MINT_PRIVATE_KEY, MINT_DATABASE_URL
```
Comment (callebtc) [inline cashu/mint/startup.py:10]:
```suggestion
from cashu.core.settings import CASHU_DIR, LIGHTNING, MINT_PRIVATE_KEY, MINT_DATABASE
```

File: cashu/core/settings.py [hunk 3]
```diff
@@ -43,6 +43,7 @@
 MINT_URL = env.str("MINT_URL", default=None)
 MINT_HOST = env.str("MINT_HOST", default="8333.space")
 MINT_PORT = env.int("MINT_PORT", default=3338)
+MINT_DATABASE = env.str("MINT_DATABASE_URL", default="data/mint")
```
Comment (callebtc) [inline cashu/core/settings.py:46]:
```suggestion
MINT_DATABASE = env.str("MINT_DATABASE", default="data/mint")
```

File: cashu/mint/startup.py [hunk 4]
```diff
@@ -7,13 +7,13 @@
...
 from cashu.core.migrations import migrate_databases
-from cashu.core.settings import CASHU_DIR, LIGHTNING, MINT_PRIVATE_KEY
+from cashu.core.settings import CASHU_DIR, LIGHTNING, MINT_PRIVATE_KEY, MINT_DATABASE
 from cashu.lightning.lnbits import LNbitsWallet
 from cashu.mint import migrations
 from cashu.mint.ledger import Ledger
 
 ledger = Ledger(
-    db=Database("mint", "data/mint"),
+    db=Database("mint", MINT_DATABASE_URL),
```
Comment (callebtc) [inline cashu/mint/startup.py:16]:
```suggestion
    db=Database("mint", MINT_DATABASE),
```

**Comment by callebtc** [comment]
> psycopg2-binary

Please add `psycopg2-binary` as a **optional** dependency to pyproject.toml under the group name `"postgres"`

**Comment by callebtc** [comment]
We also need to catch the error on import if the user chose postgres without installing the dependencies:

```
try:
    import psycopg2
except ImportError:
    raise Exception("Install psycopg2 or psycopg2-binary if you want to use a PostgreSQL database")
```

**Comment by trbouma** [comment]
ok will do!

**Comment by callebtc** [comment]
Please run `make format` and push again.

**Review by trbouma — COMMENTED** [review]
ok

**Comment by trbouma** [comment]
ok - I think I added psycopg2-binary to the pull request. Can you check if ok?

**Comment by callebtc** [comment]
Hi @trbouma I've resolved the conflicts. Can you run `make format` once again please? 

I would like to put the postgres dependency as an optional one into the pyproject. Can you do that?

**Comment by callebtc** [comment]
We should also add the default entry into `.env.example` and a comment of how to use postgres and what kind of `MINT_DATABASE` entry to add if you want to use it.

**Comment by trbouma** [comment]
in other pull request

**Pull request closed without merging**